### PR TITLE
Celery 5 upgrade and compliance

### PR DESCRIPTION
**Pull request by CodePint**
Upgrading to celery 5
- [Upgrading from Celery 4.x](https://docs.celeryproject.org/en/stable/history/whatsnew-5.0.html#upgrading-from-celery-4-x)

Change Notes
- [python 3.6 is now end of life](https://docs.celeryproject.org/en/stable/changelog.html#b3)
- [Confusion regarding flower 1.0.0 supports Celery 5 (it does)](https://github.com/mher/flower/issues/1144#issuecomment-1064282547)

Package Upgrades
- celery==4.4.7 -> [celery==5.1.2](https://docs.celeryproject.org/en/master/history/changelog-5.1.html#id1) (latest supported version for python 3.6)
- flower==0.9.3 -> [flower==1.0.0](https://pypi.org/project/flower/1.0.0/) (required for celery 5)
- kombu==4.6.11 -> [kombu==5.1.0](https://docs.celeryproject.org/projects/kombu/en/stable/changelog.html#version-5-1-0) (required for celery 5)

Signed-off-by: George Eddie geddie@linius.com

### CODE REVIEW

**Comment by ncrocfer** [comment]
Thanks for this PR, we'll review & merge it next week if all our internal tests pass ;)

Meanwhile can you run `black` on the code please? (the linter failed https://github.com/ovh/celery-director/runs/5515166540?check_suite_focus=true).

**Comment by CodePint** [comment]
> Thanks for this PR, we'll review & merge it next week if all our internal tests pass ;)
> 
> Meanwhile can you run `black` on the code please? (the linter failed https://github.com/ovh/celery-director/runs/5515166540?check_suite_focus=true).

Sure thing, I'll do that now and force push

**Comment by CodePint** [comment]
> > Thanks for this PR, we'll review & merge it next week if all our internal tests pass ;)
> > Meanwhile can you run `black` on the code please? (the linter failed https://github.com/ovh/celery-director/runs/5515166540?check_suite_focus=true).
> 
> Sure thing, I'll do that now and force push

File: director/commands/celery.py [hunk 1]
```diff
@@ -57,5 +57,6 @@ def worker(dev_mode, worker_args):
 def flower(ctx, flower_args):
     """Start the flower instance"""
     broker = ctx.app.config["CELERY_CONF"]["broker_url"]
-    args = ["flower", "-b", broker] + list(flower_args)
+    args = ["celery", "-b", broker, "flower"]
+    args += list(flower_args)
```
Comment (anthonyolea) [inline director/commands/celery.py:61]: IMO, I don't think this is better.
Actually your PR should focus on the title you gave: "Celery 5 upgrade and compliance".

File: director/commands/celery.py [hunk 2]
```diff
@@ -18,9 +18,9 @@ def beat(dev_mode, beat_args):
     """Start the beat instance"""
     args = [
         "celery",
-        "beat",
         "-A",
         "director._auto:cel",
+        "beat",
```
Comment (anthonyolea) [inline director/commands/celery.py:23]: Can you explain this change please?
Is it mandatory and introduced with Celery v5?

File: director/commands/celery.py [hunk 3]
```diff
@@ -38,9 +38,9 @@ def worker(dev_mode, worker_args):
     """Start a Celery worker instance"""
     args = [
         "celery",
-        "worker",
         "-A",
         "director._auto:cel",
+        "worker",
```
Comment (anthonyolea) [inline director/commands/celery.py:43]: Same question here than the change on the "beat" command.

File: director/commands/celery.py (hunk 1 above)
Comment (CodePint) [reply director/commands/celery.py:61]: Yes, this is a mandatory change which is introduced by celery 5. 

I have linked the documentation for the breaking CLI changes here [Upgrading from Celery 4.x](https://docs.celeryproject.org/en/stable/history/whatsnew-5.0.html#upgrading-from-celery-4-x).
Unfortunately their project/docs site is currently down: https://github.com/celery/celery/issues/7351, It is however accessible using the archive on [wayback machine](https://web.archive.org/web/20210615214611/https://docs.celeryproject.org/en/stable/history/whatsnew-5.0.html#upgrading-from-celery-4-x).


- Reference to celery CLI changes in [flower 1.0.0 release notes](https://pypi.org/project/flower/)
- Previous discussion in a closed PR: https://github.com/ovh/celery-director/pull/115

File: director/commands/celery.py (hunk 2 above)
Comment (CodePint) [reply director/commands/celery.py:23]: see: https://github.com/ovh/celery-director/pull/137#discussion_r826368412

File: director/commands/celery.py (hunk 3 above)
Comment (CodePint) [reply director/commands/celery.py:43]: see: https://github.com/ovh/celery-director/pull/137#discussion_r826368412

File: director/commands/celery.py (hunk 1 above)
Comment (CodePint) [reply director/commands/celery.py:61]: related [stackoverflow question](https://stackoverflow.com/questions/67145004/the-support-for-this-usage-was-removed-in-celery-5-0-instead-you-should-use-a)

**Comment by ncrocfer** [comment]
Again sorry @CodePint for the delay between the review and the merge. I was going to merge this PR but I just discover it breaks the dev extra requirement:

```
$ pip install -e ".[dev]"
[...]
ERROR: Cannot install celery-director and celery-director[dev]==0.4.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    celery-director[dev] 0.4.0 depends on click==7.0
    black 20.8b1 depends on click>=7.1.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

The conflict is between the `click` dependency of `black` and the one we set in the `requirements.txt` file. Can you please fix your PR?

I promise to be quick this time to merge it ;)

**Comment by ncrocfer** [comment]
@CodePint we created another PR here: #147 

We keep your commit with your name and we added you as an official contributors ;) Thanks a lot for your help on this project !

**Pull request closed without merging**